sensors/gcp_message_sensor.py

```diff
@@ -0,0 +1,124 @@
+    If responded topic message is JSON - try to convert it to object for reuse inside st2.
+    """
+    TRIGGER = 'kafka.new_GCPmessage'
+    DEFAULT_GROUP_ID = 'st2gcpgroup'
+    DEFAULT_CLIENT_ID = 'st2gcpclient'
+
+    def __init__(self, sensor_service, config=None):
+        """
+        Parse config variables, set defaults.
+        """
+        super(KafkaGCPMessageSensor, self).__init__(sensor_service=sensor_service, config=config)
+        self._logger = self._sensor_service.get_logger(__name__)
+
+        gcp_message_sensor = self._config.get('gcp_message_sensor')
+        if not gcp_message_sensor:
+            raise ValueError(
+                '[KafkaGCPMessageSensor]: "gcp_message_sensor" config value is required!')
+
+        self._hosts = gcp_message_sensor.get('hosts')
+        if not self._hosts:
+            raise ValueError(
+                '[KafkaGCPMessageSensor]: "gcp_message_sensor.hosts" config value is required!')
+
+        self._topics = set(gcp_message_sensor.get('topics', []))
+        if not self._topics:
+            raise ValueError(
+                '[KafkaGCPMessageSensor]: "gcp_message_sensor.topics" \
+                 should list at least one topic!')
+
+        # set defaults for empty values
+        self._group_id = gcp_message_sensor.get('group_id') or self.DEFAULT_GROUP_ID
+        self._client_id = gcp_message_sensor.get('client_id') or self.DEFAULT_CLIENT_ID
+        self._consumer = None
+
+    def setup(self):
+        """
+        Create connection and initialize Kafka Consumer.
+        """
+        self._logger.debug('[KafkaGCPMessageSensor]: Initializing consumer ...')
+        self._consumer = KafkaConsumer(*self._topics,
+                                       client_id=self._client_id,
+                                       group_id=self._group_id,
+                                       bootstrap_servers=self._hosts,
+                                       deserializer_class=self._try_deserialize)
+        self._ensure_topics_existence()
+
+    def _ensure_topics_existence(self):
+        """
+        Ensure that topics we're listening to exist.
+
+        Fetching metadata for a non-existent topic will automatically try to create it
+        with the default replication factor and number of partitions (default server config).
+        Otherwise Kafka server is not configured to auto-create topics and partitions.
+        """
+        map(self._consumer._client.ensure_topic_exists, self._topics)
+        self._consumer.set_topic_partitions(*self._topics)
+
+    def run(self):
+        """
+        Run infinite loop, continuously reading for Kafka message bus,
+        dispatch trigger with payload data if message received.
+        """
+        self._logger.debug('[KafkaGCPMessageSensor]: Entering into listen mode ...')
+
+        for message in self._consumer:
+            message.value['payload']['message'] = message.\
+                value['payload']['message'].decode('base64')
```

Comment:
@lmarty Do you think there will be more changes for this new `KafkaGCPMessageSensor` or `base64` is the only feature needed?

I think it would be nice to just add `gcp_support: True/False`  (default `False`) or similar (`base64_decode: True/False`) for the existing sensor config + handle that logic in `KafkaMessageSensor` class to avoid code duplication and supporting 2 sensors with almost the same functionality.

WDYT?